### ISSUE TITLE
Add metricts for CDC

### ISSUE THREAD

**Issue by haaawk** [scylladb/scylla#6002]
We need to add some metrics to CDC to be able to monitor it.

The most important is the number of successful/unsuccessful CDC augmentations.

Additionally, the number of successful/unsuccessful preimage selects.

It would be also useful to have number of successful/unsuccessful CDC augmentations that required mutation splitting.

It might be also useful to track the number of CDC augmentations that:
* touch static row
* touch map
* touch set
* touch list
* touch udt
* touch range tombstone
* touch partition delete
* touch row delete

separately for successful and unsuccessful operations

**Comment by haaawk** [comment]
@amnonh would you have any other recommendation?

**Comment by amnonh** [comment]
It's better to start early with one for total and one for error than wait for a later everything.
Would you need latency reporting?

**Comment by haaawk** [comment]
It would be nice to have an average latency of preimage select and total CDC operation probably divided to successful and unsuccessful because unsuccessful can be slower.

**Comment by haaawk** [comment]
@amnonh Can we have multiple labels added to the same operation? What I mean is that one CDC operation can be touching static row, range delete and sth else at the same time so we would like to mark it as multiple labels.

**Comment by amnonh** [comment]
@haaawk You can add multiple labels, note that you should count whatever you are counting, only once.
In general, labels are used to group by so if you have label A and label B.
sum(metric) = sum(sum(metric by A)) = sum(sum(metric by B))

**Comment by haaawk** [comment]
> @haaawk You can add multiple labels, note that you should count whatever you are counting, only once.
> In general, labels are used to group by so if you have label A and label B.
> sum(metric) = sum(sum(metric by A)) = sum(sum(metric by B))

I don't understand. Did you mean second = to be +?

**Comment by amnonh** [comment]
No, I meant it doesn't matter how you sum, you should always get the same number.
Think that you have a metric for the number of writes and you add  labels per keyspace/table

so the total number of writes should be equal to the sum of the number the number of writes per table and should be equal to the the sum of the number of writes per keyspace.

**Comment by haaawk** [comment]
Hmm. How would it work with the following labels: STATIC_ROW, ROW_DELETE, UDT and following operations:
1. touches static row and udt
2. touches row delete and udt
3. touches static row

Total number of operations is 3. Total of UDT is 2. Total of row delete is 1. Total of static row is 2.

those labels are not exclusive.

**Comment by piodul** [comment]
@haaawk 

> It would be also useful to have number of successful/unsuccessful CDC augmentations that required mutation splitting.

What do we consider to be a successful augmentation? A successful write of the mutation generated by the augmentation?

How should we count successes/failures of augmentations if splitting is involved?

**Comment by amnonh** [comment]
I think we should give it some thoughts.
a naive approach is to add a boolean label like STATIC_ROW  ROW_DELETE and UDT

And then sum(touches{STATIC_ROW='true'}) = 2
sum(touches) = 3
etc'
But I'm not sure that is the best way

**Comment by haaawk** [comment]
Why is it not good? It sounds like exactly what we want. Even more STATIC_ROW_SUCCESSFUL + STATIC_ROW_UNSUCCESSFUL

**Comment by avikivity** [comment]
Not a bug, so not backporting.
